Source/XCDBLD/XCFrameworkInfo.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+
+public enum XCFrameworkPlatformVariant: String {
+	case simulator
+}
+
+public struct XCFrameworkLibrary {
+
+	public let identifier: String
+	public let path: String
+	public let supportedArchitectures: [String]
+	public let supportedPlatform: Platform
+	public let supportedSDK: SDK
+
+	public init?(_ dictionary: [String: Any]) {
+
+		guard let identifier = dictionary["LibraryIdentifier"] as? String,
+			let path = dictionary["LibraryPath"] as? String,
+			let supportedArchs = dictionary["SupportedArchitectures"] as? [String],
+			let supportPlatform = dictionary["SupportedPlatform"] as? String else {
+				return nil
+		}
+
+		let supportedPlatformVariant = dictionary["SupportedPlatformVariant"] as? XCFrameworkPlatformVariant
+		guard let supportedPlatform = Platform(platform: supportPlatform),
+			let sdk = SDK(platform: supportedPlatform, variant: supportedPlatformVariant) else {
+				return nil
+		}
+
+		self.identifier = identifier
+		self.path = path
+		self.supportedArchitectures = supportedArchs
+		self.supportedPlatform = supportedPlatform
+		self.supportedSDK = sdk
+	}
+}
+
+fileprivate extension Platform {
+
+	static var aliases: [String : Platform] {
+		["ios" : .iOS,
+		 "macos" : .macOS,
+		 "watchos" : .watchOS,
+		 "tvos" : .tvOS
+		]
+	}
+
+	init?(platform: String) {
+
+		guard let vanillaPlatform = Platform(rawValue: platform) else {
+			guard let aliasedPlatfrom = Platform.aliases[platform] else {
+				return nil
+			}
+			self = aliasedPlatfrom
+			return
+		}
+		self = vanillaPlatform
+	}
+}
+
+fileprivate extension SDK {
+
+	init?(platform: Platform, variant: XCFrameworkPlatformVariant?) {
+
+		let isSimulatorVariant = variant != nil && variant! == .simulator
+
+		switch platform {
+			case .iOS:
+				self = isSimulatorVariant ? .iPhoneSimulator : .iPhoneOS
+			case .macOS:
+				guard !isSimulatorVariant else { return nil }
+				self = .macOSX
+			case .tvOS:
+				self = isSimulatorVariant ? .tvSimulator : .tvOS
+			case .watchOS:
+				self = isSimulatorVariant ? .watchSimulator : .watchOS
+		}
+	}
+}
+
+/// Represents the information parsed from the Info.plist of an .xcframework bundle
+public struct XCFrameworkInfo {
```

Comment:
An example plist output:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>AvailableLibraries</key>
    <array>
        <dict>
            <key>LibraryIdentifier</key>
            <string>ios-x86_64-uikitformac</string>
            <key>LibraryPath</key>
            <string>FooBar.framework</string>
            <key>SupportedArchitectures</key>
            <array>
                <string>x86_64</string>
            </array>
            <key>SupportedPlatform</key>
            <string>ios</string>
            <key>SupportedPlatformVariant</key>
            <string>uikitformac</string>
        </dict>
        <dict>
            <key>LibraryIdentifier</key>
            <string>ios-armv7_arm64</string>
            <key>LibraryPath</key>
            <string>FooBar.framework</string>
            <key>SupportedArchitectures</key>
            <array>
                <string>armv7</string>
                <string>arm64</string>
            </array>
            <key>SupportedPlatform</key>
            <string>ios</string>
        </dict>
        <dict>
            <key>LibraryIdentifier</key>
            <string>ios-i386_x86_64-simulator</string>
            <key>LibraryPath</key>
            <string>FooBar.framework</string>
            <key>SupportedArchitectures</key>
            <array>
                <string>i386</string>
                <string>x86_64</string>
            </array>
            <key>SupportedPlatform</key>
            <string>ios</string>
            <key>SupportedPlatformVariant</key>
            <string>simulator</string>
        </dict>
    </array>
    <key>CFBundlePackageType</key>
    <string>XFWK</string>
    <key>XCFrameworkFormatVersion</key>
    <string>1.0</string>
</dict>
</plist>
```

Reply:
Did I mess it up?

Reply:
I don't mean that, just as a review note. 😃